src/main/scala/util/RationalCrossing.scala

```diff
@@ -7,6 +7,32 @@
 package util
 import Chisel._
 
+// A rational crossing must put registers on the slow side.
+// This trait covers the options of how/where to put the registers.
+// BEWARE: the source+sink must agree on the direction!
```

Comment:
Yes. Diplomacy makes sure you don't use it wrong and there are also asserts that detect if the slow is in the wrong side.